src/pytorch_lightning/strategies/colossalai.py

```diff
@@ -0,0 +1,294 @@
+    """
+
+    def __init__(
+        self,
+        use_chunk: bool = True,
+        chunk_size: Optional[int] = None,
+        enable_distributed_storage: bool = True,
+        placement_policy: str = "auto",
+        force_outputs_fp32: bool = False,
+        gpu_margin_mem_ratio: float = 0.0,
+        chunk_search_range: int = 64 * 1024**2,
+        chunk_search_n_grids: int = 1024,
+        min_chunk_size: Optional[int] = None,
+        initial_scale: float = 2**32,
+        min_scale: float = 1,
+        growth_factor: float = 2,
+        backoff_factor: float = 0.5,
+        growth_interval: int = 1000,
+        hysteresis: int = 2,
+        max_scale: float = 2**32,
+        accelerator: Optional["pl.accelerators.accelerator.Accelerator"] = None,
+        parallel_devices: Optional[List[torch.device]] = None,
+        cluster_environment: Optional[ClusterEnvironment] = None,
+        checkpoint_io: Optional[CheckpointIO] = None,
+        precision_plugin: Optional[PrecisionPlugin] = None,
+    ) -> None:
+        if not _COLOSSALAI_AVAILABLE:
+            raise MisconfigurationException(
+                "To use the `ColossalAIStrategy`, please install `colossalai` first. "
+                "Download `colossalai` by consulting https://colossalai.org/download."
+            )
+
+        super().__init__(
+            accelerator=accelerator,
+            parallel_devices=parallel_devices,
+            cluster_environment=cluster_environment,
+            checkpoint_io=checkpoint_io,
+            precision_plugin=precision_plugin,
+        )
+
+        self.use_chunk = use_chunk
+        self.chunk_size = chunk_size
+        self.enable_distributed_storage = enable_distributed_storage
+        self.placement_policy = placement_policy
+        self.force_outputs_fp32 = force_outputs_fp32
+        self.gpu_margin_mem_ratio = gpu_margin_mem_ratio
+        self.chunk_size_search_kwargs = {
+            "search_range": chunk_search_range,
+            "n_grids": chunk_search_n_grids,
+            "min_chunk_size": min_chunk_size,
+        }
+        self.amp_kwargs = {
+            "initial_scale": initial_scale,
+            "min_scale": min_scale,
+            "growth_factor": growth_factor,
+            "backoff_factor": backoff_factor,
+            "growth_interval": growth_interval,
+            "hysteresis": hysteresis,
+            "max_scale": max_scale,
+        }
+        self._num_nodes = 1
+        self._logger = get_dist_logger()
+
+    def setup_distributed(self):
+        if not gpc.is_initialized(ParallelMode.GLOBAL):
+            disable_existing_loggers()
+            gpc.init_global_dist(
+                rank=self.global_rank,
+                world_size=self.world_size,
+                backend="nccl",
+                host=self.cluster_environment.main_address,
+                port=self.cluster_environment.main_port,
+            )
+            gpc.set_device(self.local_rank)
+        self.process_group = ProcessGroup()
```

Comment:
this attribute isn't declared in init. 
Please note my comment earlier about overriding DDPStrategy. DDP has it's own attribute called _process_group_backend. Another reason why you might want to consider inheriting from the base class, rather than from DDP.

Reply:
I have removed this attribute.